modules/gojira-core/src/main/java/com/flipkart/gojira/external/ExternalConfigRepository.java

```diff
@@ -17,29 +17,32 @@
 package com.flipkart.gojira.external;
 
 import com.flipkart.gojira.external.config.ExternalConfig;
+import com.flipkart.gojira.models.TestDataType;
+
 import java.util.HashMap;
 import java.util.Map;
 
-/**
- * This class acts as repository interface for {@link ExternalConfig}
- */
+/** This class acts as repository interface for {@link ExternalConfig} */
 public abstract class ExternalConfigRepository {
 
-  protected static final Map<String, ExternalConfig> externalConfigHashMap = new HashMap<>();
-
+  protected static final Map<String, Map<TestDataType, ExternalConfig>> externalConfigHashMap =
```

Comment:
Can we use somthing like this instead of TestDataType Class<? extends TestDataType>?